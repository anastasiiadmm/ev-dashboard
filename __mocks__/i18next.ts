import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';

import translationEN from '~/shared/locales/translationEN.json';
import translationRU from '~/shared/locales/translationRU.json';
import translationKY from '~/shared/locales/translationKY.json';

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: { translation: translationEN },
      ru: { translation: translationRU },
      ky: { translation: translationKY },
    },
    lng: 'ru',
    fallbackLng: 'ru',
    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;
