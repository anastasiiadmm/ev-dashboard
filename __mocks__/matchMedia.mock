window.matchMedia = window.matchMedia || function(query) {
  return {
    matches: false,
    addListener: function() {},
    removeListener: function() {},
    media: query,
    onchange: null,
    dispatchEvent: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
  };
};
