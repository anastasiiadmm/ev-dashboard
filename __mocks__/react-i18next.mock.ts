jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (str) => str,
    i18n: {
      changeLanguage: () => new Promise(() => {}),
    },
  }),
  initReactI18next: {
    type: '3rdParty',
    init: () => {},
  },
  Trans: ({ children }) => children,
}));
