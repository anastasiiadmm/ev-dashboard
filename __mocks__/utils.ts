jest.mock('mobx', () => ({
  ...jest.requireActual('mobx'),
  observable: jest.fn(() => ({})),
}));
