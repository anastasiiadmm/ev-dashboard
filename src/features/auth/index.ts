export { default as Auth } from '~/features/auth/Auth/Auth';
export { default as ResetPassword } from '~/features/auth/components/ResetPassword/ResetPassword';
export { default as ChangePassword } from '~/features/auth/components/ChangePassword/ChangePassword';
