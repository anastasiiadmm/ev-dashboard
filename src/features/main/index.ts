export { default as Home } from '~/features/main/Home/Home';
