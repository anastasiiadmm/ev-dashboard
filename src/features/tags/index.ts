export { default as Tags } from './Tags/Tags';
export { default as CreateEditTagModal } from './ui/Modals/CreateEditTagModal/CreateEditTagModal';
