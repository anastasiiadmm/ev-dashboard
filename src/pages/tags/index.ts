export { TagsAsync } from './Tags/Tags.async';
export { default as CreateEditTagModal } from './ui/Modals/CreateEditTagModal/CreateEditTagModal';
