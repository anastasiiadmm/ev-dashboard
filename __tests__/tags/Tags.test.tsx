import React from 'react';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom";
import { BrowserRouter } from 'react-router-dom';

import '../../__mocks__/react-i18next.mock';
import '../../__mocks__/matchMedia.mock';
import '../../__mocks__/i18nextMock';
import { Tags } from "../../src/features/tags";

beforeAll(() => {
  process.env.NODE_ENV = 'http://localhost/:8000/';
});

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: key => key })
}));

beforeEach(() => {
  cleanup();
});

jest.mock('~/shared/utils/config', () => ({
  apiURL: 'http://localhost/:8000/',
}));

afterEach(() => {
  jest.restoreAllMocks();
});

describe('Tags UI Component', () => {
  test('Render component toMatchSnapshot()', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Tags />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
