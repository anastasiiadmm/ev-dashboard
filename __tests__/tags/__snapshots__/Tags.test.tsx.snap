// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags UI Component Render component toMatchSnapshot() 1`] = `
<DocumentFragment>
  <div
    class="ant-row ant-row-space-between Tags css-dev-only-do-not-override-qgg3xn"
    data-testid="auth-component"
  >
    <div
      class="ant-row Tags__search-pagination-block css-dev-only-do-not-override-qgg3xn"
    >
      <div>
        <form
          class="ant-form ant-form-horizontal css-dev-only-do-not-override-qgg3xn"
        >
          <div
            class="ant-form-item undefined custom-label css-dev-only-do-not-override-qgg3xn"
            style="width: 300px;"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-qgg3xn"
            >
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-qgg3xn"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper css-dev-only-do-not-override-qgg3xn"
                    >
                      <span
                        class="ant-input-prefix"
                      >
                        <img
                          alt="search"
                          src="[object Object]"
                        />
                      </span>
                      <input
                        class="ant-input css-dev-only-do-not-override-qgg3xn"
                        placeholder="tags.search_by"
                        type="text"
                        value=""
                      />
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="ant-row Tags__table-block css-dev-only-do-not-override-qgg3xn"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-qgg3xn ant-btn-primary Tags__button-style"
        type="button"
      >
        <span
          class="ant-btn-icon"
        >
          <img
            alt="add"
            src="[object Object]"
          />
        </span>
        <span>
          tags.add_tag
        </span>
      </button>
      <div
        class="TableComponent"
      >
        <div
          class="ant-table-wrapper css-dev-only-do-not-override-qgg3xn"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-qgg3xn"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table ant-table-scroll-horizontal"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                    style="overflow-x: auto; overflow-y: hidden;"
                  >
                    <table
                      style="width: 950px; min-width: 100%; table-layout: auto;"
                    >
                      <colgroup>
                        <col
                          class="ant-table-selection-col"
                        />
                        <col
                          style="width: 10px;"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell ant-table-selection-column"
                            scope="col"
                          >
                            <div
                              class="ant-table-selection"
                            >
                              <label
                                class="ant-checkbox-wrapper css-dev-only-do-not-override-qgg3xn"
                              >
                                <span
                                  class="ant-checkbox ant-wave-target css-dev-only-do-not-override-qgg3xn"
                                >
                                  <input
                                    aria-label="Select all"
                                    class="ant-checkbox-input"
                                    type="checkbox"
                                  />
                                  <span
                                    class="ant-checkbox-inner"
                                  />
                                </span>
                              </label>
                            </div>
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            №
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            tags.name KG
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            tags.name RU
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            tags.name ENG
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.status
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            merchants.action
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          aria-hidden="true"
                          class="ant-table-measure-row"
                          style="height: 0px; font-size: 0px;"
                        >
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                          <td
                            style="padding: 0px; border: 0px; height: 0px;"
                          >
                            <div
                              style="height: 0px; overflow: hidden;"
                            >
                               
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0"
                          data-row-key="1"
                        >
                          <td
                            class="ant-table-cell ant-table-selection-column"
                          >
                            <label
                              class="ant-checkbox-wrapper css-dev-only-do-not-override-qgg3xn"
                            >
                              <span
                                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-qgg3xn"
                              >
                                <input
                                  class="ant-checkbox-input"
                                  type="checkbox"
                                />
                                <span
                                  class="ant-checkbox-inner"
                                />
                              </span>
                            </label>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            1
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="Tags__tag"
                            >
                              Tag 1
                            </div>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="Tags__tag"
                            >
                              Тег 1
                            </div>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="Tags__tag"
                            >
                              Tag 1
                            </div>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <img
                              alt="status"
                              class="Tags__center-block"
                              src="[object Object]"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="Tags__tags-block"
                            >
                              <button
                                class="ant-btn css-dev-only-do-not-override-qgg3xn ant-btn-default ant-btn-icon-only Tags__add-button"
                                type="button"
                              >
                                <span
                                  class="ant-btn-icon"
                                >
                                  <img
                                    alt="plus"
                                    src="[object Object]"
                                  />
                                </span>
                              </button>
                              <button
                                class="ant-btn css-dev-only-do-not-override-qgg3xn ant-btn-default ant-btn-icon-only Tags__delete-button"
                                type="button"
                              >
                                <span
                                  class="ant-btn-icon"
                                >
                                  <img
                                    alt="plus"
                                    src="[object Object]"
                                  />
                                </span>
                              </button>
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 gray-row"
                          data-row-key="2"
                        >
                          <td
                            class="ant-table-cell ant-table-selection-column"
                          >
                            <label
                              class="ant-checkbox-wrapper css-dev-only-do-not-override-qgg3xn"
                            >
                              <span
                                class="ant-checkbox ant-wave-target css-dev-only-do-not-override-qgg3xn"
                              >
                                <input
                                  class="ant-checkbox-input"
                                  type="checkbox"
                                />
                                <span
                                  class="ant-checkbox-inner"
                                />
                              </span>
                            </label>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            2
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="Tags__tag"
                            >
                              Tag 2
                            </div>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="Tags__tag"
                            >
                              Тег 2
                            </div>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="Tags__tag"
                            >
                              Tag 2
                            </div>
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <img
                              alt="status"
                              class="Tags__center-block"
                              src="[object Object]"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="Tags__tags-block"
                            >
                              <button
                                class="ant-btn css-dev-only-do-not-override-qgg3xn ant-btn-default ant-btn-icon-only Tags__add-button"
                                type="button"
                              >
                                <span
                                  class="ant-btn-icon"
                                >
                                  <img
                                    alt="plus"
                                    src="[object Object]"
                                  />
                                </span>
                              </button>
                              <button
                                class="ant-btn css-dev-only-do-not-override-qgg3xn ant-btn-default ant-btn-icon-only Tags__delete-button"
                                type="button"
                              >
                                <span
                                  class="ant-btn-icon"
                                >
                                  <img
                                    alt="plus"
                                    src="[object Object]"
                                  />
                                </span>
                              </button>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
